action/protocol/staking/candidate_statemanager.go

```diff
@@ -0,0 +1,137 @@
+// Copyright (c) 2020 IoTeX Foundation
+// This is an alpha (internal) release and is not suitable for production. This source code is provided 'as is' and no
+// warranties are given as to title or non-infringement, merchantability or fitness for purpose and, to the extent
+// permitted by law, all liability for your use of the code is disclaimed. This source code is governed by Apache
+// License 2.0 that can be found in the LICENSE file.
+
+package staking
+
+import (
+	"github.com/pkg/errors"
+
+	"github.com/iotexproject/iotex-address/address"
+
+	"github.com/iotexproject/iotex-core/action/protocol"
+	"github.com/iotexproject/iotex-core/state"
+)
+
+type (
+	// CandidateStateManager is candidate manager on top of StateMangaer
+	CandidateStateManager interface {
+		protocol.StateManager
+		// candidate-related
+		Size() int
+		ContainsName(string) bool
+		ContainsOwner(address.Address) bool
+		ContainsOperator(address.Address) bool
+		ContainsSelfStakingBucket(uint64) bool
+		GetByName(string) *Candidate
+		GetByOwner(address.Address) *Candidate
+		GetBySelfStakingIndex(uint64) *Candidate
+		Upsert(*Candidate) error
+		Commit() error
```

Comment:
use committer interface

Reply:
this needs to wait until we move CandCenter into sf, at that time CandidateStateManager can obtain the center from sm